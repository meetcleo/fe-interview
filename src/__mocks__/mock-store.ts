import configureStore from 'redux-mock-store';
import thunk, { ThunkDispatch } from 'redux-thunk';

import { AppActions, AppState } from '../types/app';

type DispatchExts = ThunkDispatch<AppState, void, AppActions>;

export const mockStore = configureStore<AppState, DispatchExts>([thunk]);
