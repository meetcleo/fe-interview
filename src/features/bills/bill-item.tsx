/* eslint-disable jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions */
import React, { useState } from 'react';

import { Bill } from '../../types/bill-types';

import defaultImage from '../../assets/cleo_coin.jpg';
import styles from './bill-item.module.scss';

interface Props {
  item: Bill;
  handleUpdateBill: (bill: Bill) => void;
}

export const BillItem: React.FC<Props> = ({ item, handleUpdateBill }) => {
  const { name, isBill, transactions } = item;

  const [open, setOpen] = useState(false);

  const onClick = (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
    event.stopPropagation();

    handleUpdateBill({ ...item, isBill: !item.isBill });
  };

  return (
    <div className={styles.bill} onClick={() => setOpen(!open)}>
      <div className={styles.billSummaryContainer}>
        <div className={styles.billInfo}>
          <div className={styles.billImageContainer}>
            {/* <img src={iconUrl || defaultImage} alt={name} /> */}
            <img src={defaultImage} alt={name} />
          </div>
          <div className={styles.billTitle}>{name}</div>
        </div>
        <div className={styles.transactionAmount}>
          <div className={styles.amount}>{transactions.length}</div>
        </div>
      </div>
      <div className={styles.updateBill} onClick={onClick}>
        {isBill ? 'Remove bill' : 'Add as bill'}
      </div>
      {open && (
        <div className={styles.transactions}>
          {transactions.map((transaction) => {
            const { amount, date, id } = transaction;

            return (
              <div key={id} className={styles.transactionContainer}>
                <div>Â£{amount}</div>
                <div>{date}</div>
              </div>
            );
          })}
        </div>
      )}
    </div>
  );
};
