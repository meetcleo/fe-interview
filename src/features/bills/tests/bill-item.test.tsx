import { render } from 'enzyme';
import React from 'react';

import { MOCK_BILLS } from '../../../__mocks__/mock-factory';
import { BillItem, Props } from '../bill-item';

describe('<BillItem />', () => {
  it('renders', () => {
    const mockProps: Props = {
      item: MOCK_BILLS[0],
      handleUpdateBill: jest.fn(),
    };
    const wrapper = render(<BillItem {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
