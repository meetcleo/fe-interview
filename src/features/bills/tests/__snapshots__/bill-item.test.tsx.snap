// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BillItem /> renders 1`] = `
<div
  class="bill"
>
  <div
    class="billSummaryContainer"
  >
    <div
      class="billInfo"
    >
      <div
        class="billImageContainer"
      >
        <img
          alt="Vodafone"
          src="https://pbs.twimg.com/profile_images/915824582455627776/nsyHCFq9.jpg"
        />
      </div>
      <div
        class="billTitle"
      >
        Vodafone
      </div>
    </div>
    <div
      class="transactionAmount"
    >
      <div
        class="amount"
      >
        5
      </div>
    </div>
  </div>
  <div
    class="updateBill"
  >
    Add as bill
  </div>
</div>
`;
